src/_guides/language/effective-dart/design.md

```diff
@@ -1151,6 +1212,90 @@ for one or more of the function's parameters. In those cases, you may need to
 annotate.
 
 
+### AVOID type annotating generic invocations that can be inferred.
+
+It's often useful to annotate *declarations* because they form the boundary of your API. But when instantiating an object or invoking a generic method, the type is often just noise. If inference would produce the correct type, omit the annotation and let it do it's job.
+
+When this rule is followed consistently, it helps draw the reader's attention to cases where inference does *not* produce the right type. The presence of the type argument sends a signal that that type is not otherwise pinned down by the surrounding code, and is telling you something useful.
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (generic-invocation)"?>
+{% prettify dart %}
+var strings = ["str", "ing"];
+var threes = new List.filled(3, 2);
+var pointThrees = threes.map((i) => i * 0.1);
+{% endprettify %}
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (generic-invocation)"?>
+{% prettify dart %}
+var strings = <String>["str", "ing"];
+var threes = new List<int>.filled(3, 2);
+var pointThrees = threes.map<double>((i) => i * 0.1);
+{% endprettify %}
+
+
+### PREFER type annotating declarations for top-level variables and fields that can't be inferred.
+
+Sometimes the initializer expression doesn't contain enough information to fully
+infer the type of a declaration and you need to give it a nudge. You often have
```

Comment:
I'd drop "and you need to give it a nudge", in part because "it" seems to be referring to the wrong thing.

Reply:
Done.